lib/jobs/dell-wsman-configure-bios.js

```diff
@@ -0,0 +1,151 @@
+
+module.exports = DellWsmanConfigureBiosFactory;
+di.annotate(DellWsmanConfigureBiosFactory, new di.Provide('Job.Dell.Wsman.ConfigureBios'));
+di.annotate(DellWsmanConfigureBiosFactory, new di.Inject(
+    'Job.Base',
+    'Logger',
+    'Promise',
+    'Assert',
+    'Util',
+    'Services.Waterline',
+    'Services.Configuration',
+    '_',
+    'HttpTool',
+    'Errors',
+    'Services.Encryption',
+    'validator'
+));
+
+function DellWsmanConfigureBiosFactory(
+    BaseJob,
+    Logger,
+    Promise,
+    assert,
+    util,
+    waterline,
+    configuration,
+    _,
+    HttpTool,
+    errors,
+    encryption,
+    validator
+) {
+    var logger = Logger.initialize(DellWsmanConfigureBiosFactory);
+
+    function DellWsmanConfigureBiosJob(options, context, taskId) {
+        DellWsmanConfigureBiosJob.super_.call(this, logger, options, context, taskId);
+        assert.object(this.options);
+        this.nodeId = this.context.target;
+    }
+
+    util.inherits(DellWsmanConfigureBiosJob, BaseJob);
+
+    DellWsmanConfigureBiosJob.prototype._run = function () {
+        var self = this;
+        self.dell = configuration.get('dell');
+
+        if (!self.dell || !self.dell.services || !self.dell.services.configuration) {
+            throw new errors.NotFoundError('Dell SCP  web service is not defined in smiConfig.json.');
+        }
+        return Promise.resolve(self.checkOBM())
+            .then(function(obm){
+                return self.configureBios(obm);
+            })
+            .then(function(){
+                return self._done();
+            })
+            .catch(function(err) {
+                self._done(err);
+            });
+    };
+
+    DellWsmanConfigureBiosJob.prototype.checkOBM = function() {
+        var self = this;
+        logger.info('checkOBM: Self.nodeID: ' + self.nodeId);
+        return waterline.nodes.findByIdentifier(self.nodeId)
+            .then(function(result) {
+                self.nodeType = result.type;
+                if (self.nodeType !== 'compute') {
+                    logger.info('SCP ConfigureBios is not applicable to node type ' + self.nodeType);
+                    self.cancel();
+                    return;
+                }
+                return waterline.obms.findByNode(self.nodeId, 'dell-wsman-obm-service', true);
+            }).then(function(obm) {
+                if (!obm) {
+                    throw new errors.NotFoundError('Cannot find DELL WSMAN OBM settings');
+                }
+                return obm;
+            });
+    };
+
+    DellWsmanConfigureBiosJob.prototype.configureBios = function(obm) {
+
+        if (!validator.isIP(obm.config.host)) {
+            throw new Error('Invalid ServerIP');
+        }
+
+        var self = this;
+        var data = {
+            "serverRequest": {
+                "serverIP": obm.config.host,
+                "serverUsername": obm.config.user,
+                "serverPassword": encryption.decrypt(obm.config.password),
```

Comment:
Since you have reveal the password in line 79, you don't need to decrypt it again.